NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests/AddApplicationInsightsTelemetry_TelemetryProcessors.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Linq;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+#if NETCOREAPP
+using Microsoft.ApplicationInsights.Extensibility.EventCounterCollector;
+#endif
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Test;
+using Microsoft.Extensions.Options;
+
+using Xunit;
+
+namespace Microsoft.ApplicationInsights.AspNetCore.Tests.Extensions.ApplicationInsightsExtensionsTests
+{
+    public class AddApplicationInsightsTelemetry_TelemetryProcessors : BaseTestClass
+    {
+        [Fact]
+        public static void VerifyCanAddInstanceOfTelemetryProcessor_UsingFactory()
+        {
+            var testConfig = new MyTelemetryProcessorConfiguration
+            {
+                IntValue = 123,
+                BoolValue = true
+            };
+
+            // SETUP
+            var services = GetServiceCollectionWithContextAccessor();
+            services.AddSingleton<ITelemetryProcessorFactory>(new MyTelemetryProcessorFactory(testConfig));
+            services.AddApplicationInsightsTelemetry(new ConfigurationBuilder().Build());
+
+            // ACT
+            IServiceProvider serviceProvider = services.BuildServiceProvider();
+            var telemetryConfiguration = serviceProvider.GetTelemetryConfiguration();
+
+            // ASSERT
+            var telemetryProcessor = telemetryConfiguration.DefaultTelemetrySink.TelemetryProcessors.OfType<MyTelemetryProcessor>().FirstOrDefault();
+            Assert.NotNull(telemetryProcessor);
+            Assert.Same(testConfig, telemetryProcessor.Configuration);
+        }
+
+        [Fact]
+        public static void VerifyCanAddInstanceOfTelemetryProcessor_UsingTelemetryConfiguration()
+        {
+            var testConfig = new MyTelemetryProcessorConfiguration
+            {
+                IntValue = 123,
+                BoolValue = true
+            };
+
+            // SETUP
+            var services = GetServiceCollectionWithContextAccessor();
+            services.AddApplicationInsightsTelemetry(new ConfigurationBuilder().Build());
```

Comment:
any reason why the AddApplicationInsightsTelemetry() was not used?

Reply:
i copied this from another unit test. i will change it